clang-ir fails to build

```
2024-11-28T00:45:59.0719519Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Attributes.h:424:49: warning: parameter 'ty' set but not used [-Wunused-but-set-parameter]
2024-11-28T00:45:59.0720505Z In file included from /root/llvm-project/llvm/../mlir/include/mlir/IR/Value.h:16,
2024-11-28T00:45:59.0721091Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/AffineMap.h:18,
2024-11-28T00:45:59.0721718Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/BuiltinAttributeInterfaces.h:12,
2024-11-28T00:45:59.0722375Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/BuiltinAttributes.h:12,
2024-11-28T00:45:59.0723045Z                  from /root/llvm-project/clang/include/clang/CIR/Dialect/IR/CIROpsEnums.h:17,
2024-11-28T00:45:59.0723655Z                  from /root/llvm-project/clang/include/clang/CIR/Dialect/IR/CIRAttrs.h:16,
2024-11-28T00:45:59.0724220Z                  from /root/llvm-project/clang/lib/CIR/Dialect/IR/CIRAttrs.cpp:13:
2024-11-28T00:45:59.0725589Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Types.h: In instantiation of 'static bool llvm::CastInfo<To, From, typename std::enable_if<(is_same_v<mlir::Type, typename std::remove_const<From>::type> || is_base_of_v<mlir::Type, From>), void>::type>::isPossible(mlir::Type) [with To = mlir::Type; From = mlir::Type]':
2024-11-28T00:45:59.0727693Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:311:29:   required from 'static To llvm::DefaultDoCastIfPossible<To, From, Derived>::doCastIfPossible(From) [with To = mlir::Type; From = mlir::Type; Derived = llvm::CastInfo<mlir::Type, mlir::Type, void>]'
2024-11-28T00:45:59.0729205Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:657:46:   required from 'decltype(auto) llvm::dyn_cast(From&) [with To = mlir::Type; From = mlir::Type]'
2024-11-28T00:45:59.0730252Z /root/build/tools/clang/include/clang/CIR/Dialect/IR/CIROpsAttributes.cpp.inc:700:62:   required from here
2024-11-28T00:45:59.0731220Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Types.h:417:44: warning: parameter 'ty' set but not used [-Wunused-but-set-parameter]
2024-11-28T00:45:59.0731916Z   417 |   static inline bool isPossible(mlir::Type ty) {
2024-11-28T00:45:59.0732303Z       |                                 ~~~~~~~~~~~^~
2024-11-28T00:45:59.0733533Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Types.h: In instantiation of 'static bool llvm::CastInfo<To, From, typename std::enable_if<(is_same_v<mlir::Type, typename std::remove_const<From>::type> || is_base_of_v<mlir::Type, From>), void>::type>::isPossible(mlir::Type) [with To = cir::ArrayType; From = const cir::ArrayType]':
2024-11-28T00:45:59.0735227Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:549:46:   required from 'bool llvm::isa(const From&) [with To = cir::ArrayType; From = cir::ArrayType]'
2024-11-28T00:45:59.0736391Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:572:3:   required from 'decltype(auto) llvm::cast(From&) [with To = cir::ArrayType; From = cir::ArrayType]'
2024-11-28T00:45:59.0737449Z /root/build/tools/clang/include/clang/CIR/Dialect/IR/CIROpsAttributes.cpp.inc:819:50:   required from here
2024-11-28T00:45:59.0738407Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Types.h:417:44: warning: parameter 'ty' set but not used [-Wunused-but-set-parameter]
2024-11-28T00:45:59.0739201Z In file included from /root/llvm-project/llvm/../mlir/include/mlir/IR/Location.h:17,
2024-11-28T00:45:59.0739772Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/UseDefLists.h:16,
2024-11-28T00:45:59.0740288Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/Value.h:17,
2024-11-28T00:45:59.0740783Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/AffineMap.h:18,
2024-11-28T00:45:59.0741381Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/BuiltinAttributeInterfaces.h:12,
2024-11-28T00:45:59.0742044Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/BuiltinAttributes.h:12,
2024-11-28T00:45:59.0742642Z                  from /root/llvm-project/clang/include/clang/CIR/Dialect/IR/CIROpsEnums.h:17,
2024-11-28T00:45:59.0743231Z                  from /root/llvm-project/clang/include/clang/CIR/Dialect/IR/CIRAttrs.h:16,
2024-11-28T00:45:59.0743767Z                  from /root/llvm-project/clang/lib/CIR/Dialect/IR/CIRAttrs.cpp:13:
2024-11-28T00:45:59.0745248Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Attributes.h: In instantiation of 'static bool llvm::CastInfo<To, From, typename std::enable_if<(is_same_v<mlir::Attribute, typename std::remove_const<From>::type> || is_base_of_v<mlir::Attribute, From>), void>::type>::isPossible(mlir::Attribute) [with To = cir::GlobalViewAttr; From = const cir::GlobalViewAttr]':
2024-11-28T00:45:59.0747094Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:549:46:   required from 'bool llvm::isa(const From&) [with To = cir::GlobalViewAttr; From = cir::GlobalViewAttr]'
2024-11-28T00:45:59.0748408Z /root/build/tools/clang/include/clang/CIR/Dialect/IR/CIROpsAttributes.cpp.inc:2443:52:   required from here
2024-11-28T00:45:59.0749489Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Attributes.h:424:49: warning: parameter 'ty' set but not used [-Wunused-but-set-parameter]
2024-11-28T00:45:59.0750279Z   424 |   static inline bool isPossible(mlir::Attribute ty) {
2024-11-28T00:45:59.0750669Z       |                                 ~~~~~~~~~~~~~~~~^~
2024-11-28T00:45:59.0752101Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Attributes.h: In instantiation of 'static bool llvm::CastInfo<To, From, typename std::enable_if<(is_same_v<mlir::Attribute, typename std::remove_const<From>::type> || is_base_of_v<mlir::Attribute, From>), void>::type>::isPossible(mlir::Attribute) [with To = cir::IntAttr; From = const cir::IntAttr]':
2024-11-28T00:45:59.0753980Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:549:46:   required from 'bool llvm::isa(const From&) [with To = cir::IntAttr; From = cir::IntAttr]'
2024-11-28T00:45:59.0755231Z /root/build/tools/clang/include/clang/CIR/Dialect/IR/CIROpsAttributes.cpp.inc:2455:48:   required from here
2024-11-28T00:45:59.0756362Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Attributes.h:424:49: warning: parameter 'ty' set but not used [-Wunused-but-set-parameter]
2024-11-28T00:45:59.0757231Z In file included from /root/llvm-project/llvm/../mlir/include/mlir/IR/Value.h:16,
2024-11-28T00:45:59.0757827Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/AffineMap.h:18,
2024-11-28T00:45:59.0758475Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/BuiltinAttributeInterfaces.h:12,
2024-11-28T00:45:59.0759199Z                  from /root/llvm-project/llvm/../mlir/include/mlir/IR/BuiltinAttributes.h:12,
2024-11-28T00:45:59.0759845Z                  from /root/llvm-project/clang/include/clang/CIR/Dialect/IR/CIROpsEnums.h:17,
2024-11-28T00:45:59.0760587Z                  from /root/llvm-project/clang/include/clang/CIR/Dialect/IR/CIRAttrs.h:16,
2024-11-28T00:45:59.0761145Z                  from /root/llvm-project/clang/lib/CIR/Dialect/IR/CIRAttrs.cpp:13:
2024-11-28T00:45:59.0762537Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Types.h: In instantiation of 'static bool llvm::CastInfo<To, From, typename std::enable_if<(is_same_v<mlir::Type, typename std::remove_const<From>::type> || is_base_of_v<mlir::Type, From>), void>::type>::isPossible(mlir::Type) [with To = mlir::Type; From = const mlir::Type]':
2024-11-28T00:45:59.0764235Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:549:46:   required from 'bool llvm::isa(const From&) [with To = mlir::Type; From = mlir::Type]'
2024-11-28T00:45:59.0765435Z /root/llvm-project/llvm/include/llvm/Support/Casting.h:566:3:   required from 'decltype(auto) llvm::cast(const From&) [with To = mlir::Type; From = mlir::Type]'
2024-11-28T00:45:59.0768068Z /root/llvm-project/llvm/../mlir/include/mlir/IR/AttrTypeSubElements.h:436:21:   required from 'static T mlir::AttrTypeSubElementHandler<T, typename std::enable_if<(is_base_of_v<mlir::Attribute, T> || is_base_of_v<mlir::Type, T>), void>::type>::replace(T, mlir::AttrSubElementReplacements&, mlir::TypeSubElementReplacements&) [with T = mlir::Type; mlir::AttrSubElementReplacements = mlir::AttrTypeSubElementReplacements<mlir::Attribute>; mlir::TypeSubElementReplacements = mlir::AttrTypeSubElementReplacements<mlir::Type>]'
2024-11-28T00:45:59.0771972Z /root/llvm-project/llvm/../mlir/include/mlir/IR/AttrTypeSubElements.h:491:57:   required from 'static auto mlir::AttrTypeSubElementHandler<std::tuple<_Tps ...>, typename std::enable_if<has_sub_attr_or_type_v<Ts ...>, void>::type>::replace(const std::tuple<_Tps ...>&, mlir::AttrSubElementReplacements&, mlir::TypeSubElementReplacements&) [with Ts = {mlir::Type}; mlir::AttrSubElementReplacements = mlir::AttrTypeSubElementReplacements<mlir::Attribute>; mlir::TypeSubElementReplacements = mlir::AttrTypeSubElementReplacements<mlir::Type>]'
2024-11-28T00:45:59.0774787Z /root/llvm-project/llvm/../mlir/include/mlir/IR/AttrTypeSubElements.h:586:70:   required from 'auto mlir::detail::replaceImmediateSubElementsImpl(T, llvm::ArrayRef<mlir::Attribute>&, llvm::ArrayRef<mlir::Type>&) [with T = cir::ZeroAttr]'
2024-11-28T00:45:59.0777449Z /root/llvm-project/llvm/../mlir/include/mlir/IR/StorageUniquerSupport.h:145:61:   required from 'mlir::detail::StorageUserBase<ConcreteT, BaseT, StorageT, UniquerT, Traits>::getReplaceImmediateSubElementsFn() [with ConcreteT = cir::ZeroAttr; BaseT = mlir::Attribute; StorageT = cir::detail::ZeroAttrStorage; UniquerT = mlir::detail::AttributeUniquer; Traits = {mlir::TypedAttr::Trait}]::<lambda(auto:7, llvm::ArrayRef<mlir::Attribute>, llvm::ArrayRef<mlir::Type>)> [with auto:7 = mlir::Attribute]'
2024-11-28T00:45:59.0783267Z /root/llvm-project/llvm/include/llvm/ADT/STLFunctionalExtras.h:62:77:   required by substitution of 'template<class Callable> llvm::function_ref<mlir::Attribute(mlir::Attribute, llvm::ArrayRef<mlir::Attribute>, llvm::ArrayRef<mlir::Type>)>::function_ref(Callable&&, std::enable_if_t<(! std::is_same<typename llvm::remove_cvref<T>::type, llvm::function_ref<mlir::Attribute(mlir::Attribute, llvm::ArrayRef<mlir::Attribute>, llvm::ArrayRef<mlir::Type>)> >::value), void>*, std::enable_if_t<(std::is_void<mlir::Attribute>::value || std::is_convertible<decltype (declval<Callable>()(declval<mlir::Attribute>(), declval<llvm::ArrayRef<mlir::Attribute> >(), declval<llvm::ArrayRef<mlir::Type> >())), mlir::Attribute>::value), void>*) [with Callable = mlir::detail::StorageUserBase<ConcreteT, BaseT, StorageT, UniquerT, Traits>::getReplaceImmediateSubElementsFn() [with ConcreteT = cir::ZeroAttr; BaseT = mlir::Attribute; StorageT = cir::detail::ZeroAttrStorage; UniquerT = mlir::detail::AttributeUniquer; Traits = {mlir::TypedAttr::Trait}]::<lambda(auto:7, llvm::ArrayRef<mlir::Attribute>, llvm::ArrayRef<mlir::Type>)>]'
2024-11-28T00:45:59.0788430Z /root/llvm-project/llvm/../mlir/include/mlir/IR/AttributeSupport.h:50:12:   required from 'static mlir::AbstractAttribute mlir::AbstractAttribute::get(mlir::Dialect&) [with T = cir::ZeroAttr]'
2024-11-28T00:45:59.0789788Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Dialect.h:330:59:   required from 'void mlir::Dialect::addAttribute() [with T = cir::ZeroAttr]'
2024-11-28T00:45:59.0795596Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Dialect.h:308:62:   required from 'void mlir::Dialect::addAttributes() [with Args = {cir::LangAttr, cir::BoolAttr, cir::ZeroAttr, cir::UndefAttr, cir::ConstArrayAttr, cir::ConstVectorAttr, cir::ConstStructAttr, cir::IntAttr, cir::FPAttr, cir::ComplexAttr, cir::ConstPtrAttr, cir::CmpThreeWayInfoAttr, cir::DataMemberAttr, cir::MethodAttr, cir::SignedOverflowBehaviorAttr, cir::GlobalViewAttr, cir::TypeInfoAttr, cir::VTableAttr, cir::StructLayoutAttr, cir::DynamicCastInfoAttr, cir::AddressSpaceAttr, cir::ASTDeclAttr, cir::ASTFunctionDeclAttr, cir::ASTCXXMethodDeclAttr, cir::ASTCXXConstructorDeclAttr, cir::ASTCXXConversionDeclAttr, cir::ASTCXXDestructorDeclAttr, cir::ASTVarDeclAttr, cir::ASTTypeDeclAttr, cir::ASTTagDeclAttr, cir::ASTRecordDeclAttr, cir::ASTExprAttr, cir::ASTCallExprAttr, cir::VisibilityAttr, cir::ExtraFuncAttributesAttr, cir::InlineAttr, cir::OptNoneAttr, cir::NoThrowAttr, cir::ConvergentAttr, cir::GlobalCtorAttr, cir::GlobalDtorAttr, cir::BitfieldInfoAttr, cir::AnnotationAttr, cir::GlobalAnnotationValuesAttr, cir::TBAAAttr, cir::OpenCLKernelMetadataAttr, cir::OpenCLKernelArgMetadataAttr, cir::OpenCLVersionAttr, cir::OpenCLKernelAttr, cir::OpenCLKernelUniformWorkGroupSizeAttr, cir::CatchUnwindAttr, cir::CatchAllAttr}]'
2024-11-28T00:45:59.0801398Z /root/llvm-project/clang/lib/CIR/Dialect/IR/CIRAttrs.cpp:669:9:   required from here
2024-11-28T00:45:59.0802298Z /root/llvm-project/llvm/../mlir/include/mlir/IR/Types.h:417:44: warning: parameter 'ty' set but not used [-Wunused-but-set-parameter]
2024-11-28T00:45:59.0803088Z   417 |   static inline bool isPossible(mlir::Type ty) {
2024-11-28T00:45:59.0803465Z       |                                 ~~~~~~~~~~~^~
2024-11-28T00:46:06.1144143Z [6057/6590] Building CXX object tools/mlir/lib/CAPI/RegisterEverything/CMakeFiles/obj.MLIRCAPIRegisterEverything.dir/RegisterEverything.cpp.o
2024-11-28T00:46:06.1145108Z ninja: build stopped: subcommand failed.
2024-11-28T00:46:10.1276318Z ##[error]Process completed with exit code 1.
2024-11-28T00:46:10.1330206Z ##[group]Run ./post-run.sh "clang_ir" "" "" "1732753735"
2024-11-28T00:46:10.1330675Z [36;1m./post-run.sh "clang_ir" "" "" "1732753735"[0m
2024-11-28T00:46:10.1345797Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
```

--- a/build/build.sh
+++ b/build/build.sh
@@ -107,6 +107,8 @@ patmat-trunk)
     LLVM_ENABLE_RUNTIMES+=";libunwind"
     ;;
 clangir-trunk)
+    # Does not compile with 9.2.0.
+    GCC_VERSION=14.2.0
     BRANCH=main
     URL=https://github.com/llvm/clangir.git
     VERSION=clangir-trunk-$(date +%Y%m%d)

